dart/dynamics/Joint.h

```diff
@@ -557,8 +620,19 @@ class Joint : public virtual common::Subject
   friend class Skeleton;
 
 protected:
+
+  /// Constructor called by inheriting class
+  Joint(const Properties& _properties);
+
+  /// Create a clone of this Joint. This may only be called by the Skeleton
+  /// class.
+  virtual Joint* clone() const = 0;
+
+  /// Called by the Skeleton class
+  virtual void registerDofs() = 0;
+
   /// Initialize this joint. This function is called by BodyNode::init()
-  virtual void init(Skeleton* _skel);
+  virtual void init(std::shared_ptr<Skeleton> _skel);
```

Comment:
That's a fair point; I thought we were assuming that no one uses it.

I guess if we deprecate it for a while, we can wait and see if anyone complains that we've messed up their custom Joint type. And if no one ever complains, *then* we can remove it.